src/pages/home/report/ReportActionsView.js

```diff
@@ -397,6 +426,68 @@ class ReportActionsView extends React.Component {
         }
     }
 
+    /**
+     * Scrolls to a specific report action ID
+     */
+    scrollToReportActionID() {
+        this.actionIndexID = _.findIndex(this.sortedReportActions, (
+            ({action}) => parseInt(action.reportActionID, 10) === this.props.reportActionID
+        ));
+
+        if (this.actionIndexID !== -1) {
+            ReportScrollManager.scrollToIndex({index: this.actionIndexID, viewPosition: 0.5});
+            this.setState({selectedReportActionID: this.props.reportActionID});
+        }
+    }
+
+    /**
+     * Records when a report actionID is done rendering.
+     *
+     * @param {string} reportActionID
+     */
+    recordReportActionIDRendered(reportActionID) {
+        this.renderedActionIDs.add(parseInt(reportActionID, 10));
+        this.checkScrollToReportAction();
+    }
+
+    /**
+     * Records when our FlatList is done measuring the heights and offset of items. Since native will hit this block immediately we'll use a param checkScrollToReportAction to prevent
+     * calling the scroll method unnecessarily until the item has been rendered.
+     *
+     * @param {boolean} checkScrollToReportAction
+     */
+    recordMeasurementDone(checkScrollToReportAction = true) {
+        this.doneMeasuring = true;
+        if (checkScrollToReportAction) {
+            this.checkScrollToReportAction();
+        }
+    }
+
+    /**
+     * Determine if we can scroll now or not.
+     * On web we can scroll only after measurements of all rendered items are done while on native we can scroll after rendering of the specific item is done.
+     */
+    checkScrollToReportAction() {
+        if (!this.props.reportActionID) {
+            return;
+        }
+
+        if (this.doneMeasuring && !this.doneScrollingToReportActionID) {
+            if (this.renderedActionIDs.has(this.props.reportActionID)) {
+                this.doneScrollingToReportActionID = true;
+
+                // We give a slight delay because if we attempt this too fast the scroll gets buggy on iOS as more items need to render.
+                setTimeout(this.scrollToReportActionID, 500);
```

Comment:
Any suggestion for removing setTimeout here? I tried removing it but the scroll breaks on native when scrolling to an older comment. I think it because its needs an extra half a second to actually render itself and the items around the one we're scrolling to.

Reply:
Alternative could be...

- `ReportActionsList` tells this view that all the actions are loaded with a callback 
- This view will set a flag so that it will know it has to scroll somewhere when the callback fires

```js
onAllItemsLayout={() => {
    if (!this.props.reportActionID || ! this.shouldScrollToReportActionID) {
        return;
    }

    this.scrollToReportActionID();
}}
```

Reply:
> ReportActionsList tells this view that all the actions are loaded with a callback

This would be really nice but i don't know how to do this. SortedReportActions can contain 150 values but on mobile just 20 or 30 would be rendered so how can ReportActionsList know when all 20 or 30 are rendered and no more are being rendered atm?

Reply:
Cool, how do we do this part in code?
> Items render and we track idleness of rendering 

Also mind if i note this down as part of future improvement and leave it as is for now?

Reply:
There may be a way to get rendered item count from native, we certainly can't be the first people to need to know this. 

Also, if we end up keeping the timeout, could we at least reduce it a bit? 1/2 a sec is a long time.

Reply:
Cool i'll bring up this discussion tomorrow morning in #engg-chat to discuss a solution.